### ISSUE TITLE
[dev.icinga.com #10097] Hisotry missing service/host state

### ISSUE THREAD

**Issue by icinga-migration** [Icinga/icingaweb2#1984]
This issue has been migrated from Redmine: https://dev.icinga.com/issues/10097

**Created by arioch on 2015-09-04 12:07:07 +00:00**

Assignee: _(none)_
Status: _Closed_  (closed on _2016-02-16 12:04:03 +00:00_)
Target Version: _(none)_
Last Update: _2016-02-16 12:04:03 +00:00 (in Redmine)_

---

Hi,

Couldn't seem to find any documentation regarding this subject but the history tab per host or service does not show any activity other than scheduled downtime, acknowledgements and comments.
I'd like to see the service or host states too. Is this configurable? I'd expect this to be enabled by default...

The information is readily available when I check the database content. It just doesn't show up in Web2.

    [root@icinga ~]# mysql -u root -p -e 'select * from icinga_statehistory limit 10;'
    statehistory_id instance_id state_time  state_time_usec object_id   state_change    state   state_type  current_check_attempt   max_check_attempts  last_state  last_hard_state output  long_output check_source    endpoint_object_id
    1   1   2015-01-06 11:42:49 51947   61  1   0   1   1   3   3   0   USERS OK - 1 users currently logged in      icinga.example.com  1
    2   1   2015-01-06 11:42:53 272351  58  1   0   1   1   3   3   0   OK - load average: 0.04, 0.03, 0.09     icinga.example.com  1
    3   1   2015-01-06 11:43:00 270935  52  1   2   1   1   3   3   2   CRITICAL - Network Unreachable (::1)        icinga.example.com  1
    4   1   2015-01-06 11:43:02 651293  56  1   0   1   1   3   3   0   DISK OK - free space: / 12360 MB (67% inode=99%):       icinga.example.com  1
    5   1   2015-01-06 11:43:09 753046  55  1   0   1   1   3   3   0   DISK OK - free space: / 12360 MB (67% inode=99%): /dev 485 MB (100% inode=99%): /dev/shm 493 MB (100% inode=99%): /run 443 MB (89% inode=99%): /sys/fs/cgroup 493 MB (100% inode=99%): /boot 859 MB (84% inode=99%):        icinga.example.com  1
    6   1   2015-01-06 11:43:10 111594  53  1   0   1   1   3   3   0   SSH OK - OpenSSH_6.4 (protocol 2.0)         icinga.example.com  1
    7   1   2015-01-06 11:43:14 113390  48  1   0   1   1   5   1   0   PING OK - Packet loss = 0%, RTA = 0.04 ms       icinga.example.com  1
    8   1   2015-01-06 11:43:20 505104  51  1   0   1   1   3   3   0   PING OK - Packet loss = 0%, RTA = 0.04 ms       icinga.example.com  1
    9   1   2015-01-06 11:43:22 945935  59  1   0   1   1   3   3   0   PROCS OK: 235 processes         icinga.example.com  1

Relevant parts from /etc/icinga2/features-enabled/ido-mysql.conf:

    object IdoMysqlConnection "mysql-ido" {
      *snip*
      cleanup = {
        acknowledgements_age = 0
        commenthistory_age = 0
        contactnotificationmethods_age = 0
        contactnotifications_age = 0
        downtimehistory_age = 0
        eventhandlers_age = 0
        externalcommands_age = 0
        flappinghistory_age = 0
        hostchecks_age = 0
        logentries_age = 0
        notifications_age = 0
        processevents_age = 0
        servicechecks_age = 0
        statehistory_age = 0
        systemcommands_age = 0
      }

Relevant parts from /etc/icingaweb2/resources.ini:

    [icinga_ido]
    type                = "db"
    db                  = "mysql"
    host                = "localhost"
    port                = "3306"
    dbname              = "icinga"
    username            = *
    password            = *

Relevant parts from /etc/icingaweb2/modules/monitoring/backends.ini:

    [icinga]
    type                = "ido"
    resource            = "icinga_ido"

Versions:

    # rpm -qa | grep -i icinga
    icinga2-2.3.9-1.el7.centos.x86_64
    icingaweb2-vendor-lessphp-0.4.0-1.el7.centos.noarch
    icinga2-common-2.3.9-1.el7.centos.x86_64
    icinga2-ido-mysql-2.3.9-1.el7.centos.x86_64
    icingaweb2-common-2.0.0-4.rc1.el7.centos.noarch
    icingaweb2-2.0.0-4.rc1.el7.centos.noarch
    icingaweb2-vendor-Parsedown-1.0.0-1.el7.centos.noarch
    python-icinga2-2.1.1-1.el7.centos.x86_64
    icinga2-ido-pgsql-2.3.9-1.el7.centos.x86_64
    php-Icinga-2.0.0-4.rc1.el7.centos.noarch
    icingaweb2-vendor-dompdf-0.6.1-1.el7.centos.noarch
    icingaweb2-vendor-Zend-1.12.9-1.el7.centos.noarch
    icingaweb2-vendor-JShrink-1.0.1-1.el7.centos.noarch
    icingaweb2-vendor-HTMLPurifier-4.6.0-1.el7.centos.noarch
    icinga2-bin-2.3.9-1.el7.centos.x86_64
